conans/client/cmd/new.py

```diff
@@ -315,17 +255,11 @@ def _get_files_from_template_dir(template_dir, name, version, package_name, defi
     return out_files
 
 
-def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False, bare=False,
+def cmd_new(name, version, header=False, pure_c=False, test=False, exports_sources=False, bare=False,
             gitignore=None, template=None, cache=None, defines=None):
-    try:
-        name, version, user, channel, revision = get_reference_fields(ref, user_channel_input=False)
-        # convert "package_name" -> "PackageName"
-        package_name = re.sub(r"(?:^|[\W_])(\w)", lambda x: x.group(1).upper(), name)
-    except ValueError:
-        raise ConanException("Bad parameter, please use full package name,"
-                             "e.g.: MyLib/1.2.3@user/testing")
 
-    # TODO: Validate it is a valid reference
+    # convert "package_name" -> "PackageName"
+    package_name = re.sub(r"(?:^|[\W_])(\w)", lambda x: x.group(1).upper(), name)
```

Comment:
Should we have this pattern "compiled" in another common file?

Reply:
It doesn't seem to have enough entity or relevance, quite specific for this use case. Better local than excessive premature generalization.